aws/corehandlers/handlers.go

```diff
@@ -106,6 +106,22 @@ var SendHandler = request.NamedHandler{
 			sender = sendWithoutFollowRedirects
 		}
 
+		if request.NoBody == r.HTTPRequest.Body {
```

Comment:
Having this here is to ensure that the exposer of the nil `HTTPRequest.Body` value doesn't escape where it is absolutely needed. Doing this in `request.Send` would allow the nil value to be exposed to other request handlers within the `Send` handlers list.